### DISCUSSION TITLE
Testing approach for import/export

### DISCUSSION THREAD

**Discussion by eliandoran** [TriliumNext/Trilium#3464] [General]
Greetings, @zadam .

I was looking through the source code for import and export as I was interested in adding support for a new import format (OnePlus Notes) and I was wondering what's the testing procedure here.

Are there any units tests or a way to run the import without having to do it directly from the UI? How do you generally test these features during development?

Thank you for your information,

Elian D.

**Comment by zadam** [comment]
Hello, I am actually testing from UI. I have a private repository of test files (private because it's composed mainly of files contributed by users with various levels of sensitivity).

TBH I've never heard of this app and I think a more suitable approach would be to create a separate conversion application into HTML which could then be imported into Trilium. I don't really plan to support many import format, ENEX is rather an exception given how many users there are.
